test/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics/AtmosphericDrag.test.cpp

```diff
@@ -221,41 +235,70 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_Dynamics_AtmosphericDrag, Ge
     EXPECT_TRUE(atmosphericDynamics.getSatelliteSystem() == satelliteSystem_);
 }
 
-TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_Dynamics_AtmosphericDrag, ApplyContribution)
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_Dynamics_AtmosphericDrag, GetReadCoordinatesSubsets)
 {
-    NumericalSolver::StateVector dxdt(6);
-    dxdt.setZero();
     AtmosphericDrag atmosphericDrag(earthSPtr_, satelliteSystem_);
-    atmosphericDrag.applyContribution(startStateVector_, dxdt, startInstant_);
-    EXPECT_GT(1e-15, 0.0 - dxdt[0]);
-    EXPECT_GT(1e-15, 0.0 - dxdt[1]);
-    EXPECT_GT(1e-15, 0.0 - dxdt[2]);
-    EXPECT_GT(1e-15, 0.0 - dxdt[3]);
-    EXPECT_GT(5e-11, -0.0000278707803890 - dxdt[4]);
-    EXPECT_GT(5e-11, -0.0000000000197640 - dxdt[5]);
+
+    const Array<Shared<const CoordinatesSubset>> subsets = atmosphericDrag.getReadCoordinatesSubsets();
+
+    EXPECT_EQ(2, subsets.size());
+    EXPECT_EQ(CartesianPosition::Default(), subsets[0]);
+    EXPECT_EQ(CartesianVelocity::Default(), subsets[1]);
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_Dynamics_AtmosphericDrag, GetWriteCoordinatesSubsets)
+{
+    AtmosphericDrag atmosphericDrag(earthSPtr_, satelliteSystem_);
+
+    const Array<Shared<const CoordinatesSubset>> subsets = atmosphericDrag.getWriteCoordinatesSubsets();
+
+    EXPECT_EQ(1, subsets.size());
+    EXPECT_EQ(CartesianVelocity::Default(), subsets[0]);
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_Dynamics_AtmosphericDrag, ComputeContribution)
+{
+    AtmosphericDrag atmosphericDrag(earthSPtr_, satelliteSystem_);
+
+    const VectorXd contribution =
+        atmosphericDrag.computeContribution(startInstant_, startStateVector_.segment(0, 6), Frame::GCRF());
+
+    EXPECT_EQ(3, contribution.size());
+    EXPECT_GT(1e-15, 0.0 - contribution[0]);
+    EXPECT_GT(5e-11, -0.0000278707803890 - contribution[1]);
+    EXPECT_GT(5e-11, -0.0000000000197640 - contribution[2]);
 }
 
 // Test data gathered from Orekit
 TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_System_Dynamics_AtmosphericDrag, OneStepAtmosphereOnly)
```

Comment:
As mentioned to @vishwa2710 
Challenging the existence of these tests as they go above and beyond the `Dynamics` implementation.

As of now they are red, despite the `Dynamics` being fine when computing contribution. So it's a "false negative".

I've migrated them for 3/4 clases. But I don't know if we should have them.

Reply:
Agreed to remove them.